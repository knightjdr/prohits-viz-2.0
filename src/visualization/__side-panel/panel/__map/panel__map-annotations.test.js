import React from 'react';
import ShortID from 'shortid';
import { shallow } from 'enzyme';

import Annotations from './panel__map-annotations';

jest.mock('shortid');
ShortID.mockReturnValueOnce('a')
  .mockReturnValueOnce('b')
  .mockReturnValueOnce('c')
  .mockReturnValueOnce('d');

const annotations = {
  color: '#0000ff',
  list: [
    { text: 'a', x: 0.1, y: 0.2 },
    { text: 'b', x: 0.2, y: 0.4 },
  ],
};
const markers = {
  color: '#000000',
  list: [
    {
      height: 0.1,
      width: 0.1,
      x: 0.1,
      y: 0.1,
    },
    {
      height: 0.2,
      width: 0.3,
      x: 0.5,
      y: 0.5,
    },
  ],
};

describe('Map panel annotations', () => {
  it('should render', () => {
    const wrapper = shallow(
      <Annotations
        annotations={annotations}
        markers={markers}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
