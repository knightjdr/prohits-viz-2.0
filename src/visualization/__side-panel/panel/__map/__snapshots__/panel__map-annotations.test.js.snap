// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map panel annotations should render and match snapshot 1`] = `
<React.Fragment>
  <div
    className="panel__map-annotation"
    key="mapannotation1"
    style={
      Object {
        "backgroundColor": "#0000ff",
        "left": "calc(10% - 5px)",
        "top": "calc(20% - 5px)",
      }
    }
    tooltip="a"
    tooltip-position="top"
  />
  <div
    className="panel__map-annotation"
    key="mapannotation1"
    style={
      Object {
        "backgroundColor": "#0000ff",
        "left": "calc(20% - 5px)",
        "top": "calc(40% - 5px)",
      }
    }
    tooltip="b"
    tooltip-position="top"
  />
</React.Fragment>
`;
